Packet++/header/TpktLayer.h

```diff
@@ -0,0 +1,126 @@
+#ifndef PACKETPP_TPKT_LAYER
+#define PACKETPP_TPKT_LAYER
+
+#include "EthLayer.h"
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @struct tpkthdr
+	 * Represents a TPKT protocol header
+	 */
+#pragma pack(push, 1)
+	typedef struct
+	{
+		/** message version */
+		uint8_t version;
+		/** message reserved */
+		uint8_t reserved;
+		/** message length */
+		uint16_t length;
+	} tpkthdr;
+#pragma pack(pop)
+
+	/**
+	 * @class TpktLayer
+	 * Represents a TPKT (Transport Service on top of the TCP) protocol layer
+	 */
+	class TpktLayer : public Layer
+	{
+	  public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to @ref tpkthdr)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		TpktLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+			: Layer(data, dataLen, prevLayer, packet)
+		{
+			m_Protocol = TPKT;
+		}
+
+		TpktLayer(uint8_t version, uint16_t length);
+
+		virtual ~TpktLayer() {}
+
+		/**
+		 * Get a pointer to the TPKT header. Data can be retrieved through the
+		 * other methods of this layer. Notice the return value points directly to the data, so every change will change
+		 * the actual packet data
+		 * @return A pointer to the @ref tpkthdr
+		 */
+		tpkthdr *getTpktHeader() const { return (tpkthdr *)m_Data; }
+
+		/**
+		 * @return TPKT reserved
+		 */
+		uint8_t getReserved() const;
+
+		/**
+		 * @return TPKT version
+		 */
+		uint8_t getVersion() const;
+
+		/**
+		 * @return TPKT length
+		 */
+		uint16_t getLength() const;
+
+		/**
+		 * @return Size of @ref tpkthdr
+		 */
+		size_t getHeaderLen() const override { return sizeof(tpkthdr); }
+
+		/**
+		 * Does nothing for this layer
+		 */
+		void computeCalculateFields() override {}
+
+		/**
+		 * Currently identifies the following next layer: PayloadLayer
+		 */
```

Comment:
`PayloadLayer` is a generic payload layer, which is another way to say "we don't know what is the next layer" 😃 

Maybe we can change the comment to something like: "Currently parses the rest of the packet as a generic payload (PayloadLayer)"